splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/RumConstants.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.rum.internal;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.common.AttributeKey;
+
+public class RumConstants {
+
+    public static final String OTEL_RUM_LOG_TAG = "OpenTelemetryRum";
+
+    public static final AttributeKey<String> COMPONENT_KEY = AttributeKey.stringKey("component");
```

Comment:
Would it be possible to leave the `component` attribute in the "Splunk space"? Kinda like I did for crash reporting: `.addAttributesExtractor(constant(COMPONENT_KEY, COMPONENT_ERROR))`

I'm wondering if the OTel spec will include this attribute.

Reply:
Oh yeah great point. I missed that API usage earlier....so this is a good idea.

As far as spec goes, that's hard to say. I feel like right now it might be kinda superfluous and can often be extrapolated from other things (span name and existence of other attributes)....but I also don't fully grok how our backend/ui might be using it.

Reply:
Ok, I managed to pull the `COMPONENT_KEY` back into the Splunk namespace. One key difference is that the activity/startup/fragment stuff uses the SDK more directly and does not go through the instrumentation APIs like the other code....

So rather than leveraging the `AttributesExtractor` like before, I wrapped the injected tracer. Please have a look at tell me what you think. If the only use case was to append a vendor-specific attribute (component), then passing an additional attribute or attributes seemed like overkill.

This exercise has me thinking more about the future in otel when many of these spans almost certainly turn into events. When that happens, I suspect that the existing instrumentation API won't serve us as well because it's pretty span-centric.

Reply:
> This exercise has me thinking more about the future in otel when many of these spans almost certainly turn into events. When that happens, I suspect that the existing instrumentation API won't serve us as well because it's pretty span-centric.

Hmm, yeah, the existing instrumentation api assumes that things have a start and an end, and that they take time. I think we'll have to ensure that whatever OTel SIG comes up with is extensible enough to pass some extra attributes in.